src/iterator.rs

```diff
@@ -350,11 +397,92 @@ impl<'a> Iterator<'a> {
 
             // Reverse direction.
             let next_ts = get_ts(self.table_iter.key());
-            let key_without_ts = user_key(self.table_iter.key());
-            if next_ts <= self.read_ts && key_without_ts == item.key {
+            if next_ts <= self.read_ts && crate::util::same_key(self.table_iter.key(), key) {
+                // This is a valid potential candidate.
+                continue;
+            }
+
+            // Ignore the next candidate. Return the current one.
+            self.item = Some(item);
+            return true;
+        }
+    }
+
+    /// Just like `parse_item`, but used for `prev`.
+    ///
+    /// This function advances the iterator.
+    fn parse_item_for_prev(&mut self) -> bool {
```

Comment:
can we reuse `parse_item` by passing a parameter indicating the direction?

Reply:
Yes, we can combine `parse_item` and `parse_item_for_prev` once the logic is reviewed.